control_protocol.md

```diff
@@ -0,0 +1,293 @@
+
+:::{note}
+TBD: Respond to every non empty message with an empty one?
+:::
+
+
+(sign-out)=
+##### Signing out
+
+A Component should tell a Coordinator, when it stops working, with a SIGNOUT message.
+The Coordinator shall ACKNOWLEDGE the sign out and remove the ID from its address book.
+It shall also publish to the other Coordinators in the network that this Component signed out, see {ref}`Coordinator coordination<coordinator-coordination>`.
+
+:::{mermaid}
+sequenceDiagram
+    CA ->> Co1: COORDINATOR|Co1.CA|SIGNOUT
+    Co1 ->> CA: Co1.CA|Co1.COORDINATOR|ACKNOWLEDGE
+    Note right of Co1: Removes "CA" with identity "IA"<br> from address book
+    Note left of CA: Shall not send any message anymore
+:::
+
+
+#### Communication with other Components
+
+The following two examples show, how a message is transferred between two components `CA`, `CB` via one or two Coordinators.
+Coordinators shall send messages from their DEALER socket to other Coordinator's ROUTER socket.
+
+Coordinators shall hand on the message to the corresponding Coordinator or connected Component.
+
+
+:::{mermaid}
+sequenceDiagram
+    CA ->> Co1: Co1.CB|Co1.CA| Give me property A.
+    Co1 ->> CB: Co1.CB|Co1.CA| Give me property A.
+    Note left of CB: Reads property A
+    CB ->> Co1: Co1.CA|Co1.CB| Property A has value 5.
+    Co1 ->> CA: Co1.CA|Co1.CB| Property A has value 5.
+    Note over CA,Co1: The first message could be as well:
+    CA ->> Co1: CB|Co1.CA| Give me property A.
+:::
+
+
+:::{mermaid}
+sequenceDiagram
+    CA ->> Co1: Co2.CB|Co1.CA| Give me property A.
+    Note over Co1,Co2: Co1.DEALER socket sends to Co2.ROUTER
+    Co1 ->> Co2: Co2.CB|Co1.CA| Give me property A.
+    Co2 ->> CB: Co2.CB|Co1.CA| Give me property A.
+    Note left of CB: Reads property A
+    CB ->> Co2: Co1.CA|Co2.CB| Property A has value 5.
+    Note over Co1,Co2: Co2.DEALER socket sends to Co1.ROUTER
+    Co2 ->> Co1: Co1.CA|Co2.CB| Property A has value 5.
+    Co1 ->> CA: Co1.CA|Co2.CB| Property A has value 5.
+:::
+
+Prerequisite of Communication between two Components are:
+- Both Components are connected to a Coordinator and {ref}`signed in<sign-in>`.
+- Both Components are either connected to the same Coordinator (example one), or their Coordinators are connected to each other (example two).
+
+
+The following flow chart shows the decision scheme and message modification in a Coordinator.
+`Node0`, `NodeR` are placeholders for sender and recipient Node ID.
+`Recipient` is a placeholder for the recipient Component ID.
+`IA` is the connection identity of `Co1.CA` or of `Node0.COORDINATOR`.
+`IB` is the connection identities of `Co1.Recipient`.
+Bold arrows indicate message flow, thin lines indicate decision flow.
+Thin, dotted lines indicate decision flow in case of errors
+
+:::{mermaid}
+flowchart TB
+    C([Node0.CA DEALER]) == "NodeR.Recipient|Node0.CA|Content" ==> R0
+    R0[receive] == "IA|NodeR.Recipient|Node0.CA|Content" ==> CN0{Node0 == Co1}
```

Comment:
"Node0 == Co1": you mean "Co1 is in Node0", right? took me a bit to parse that.

Reply:
Also, a logic question? How can the answer to that ever be "no"? By definition (elsewhere), Node0.CA's dealer socket may only connect to Node0's Coordinator Co1, so "Node0==Co1" _always has to be true, no?

Reply:
The scheme started with a locally connected Component.
Then I noticed, that the Router socket might accept a message from another Coordinator and I adjusted everything, except for the inconsistency, that the Dealer socket might belong to Node0's Coordinator, not to CA directly...

> Node0 == Co1": you mean "Co1 is in Node0", right? took me a bit to parse that.

I meant, that the placeholder Node0 (which could be anything) equals the Node ID "Co1". Code example:
~~~python
Node0, ID = sender.split(",")
If Node0 == "Co1":
   ... 
~~~

Reply:
How can I increase understandibility?

Reply:
you reworked the whole thing, I'll check it out again and give feedback on the updated text.